src/error.rs

```diff
@@ -0,0 +1,57 @@
+use thiserror::Error;
+
+use crate::stream::Stream;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("index_over_kdr > 0 is not supported yet")]
+    UnsupportedIndexOverKdr,
+    #[error("SRTP Master Key must be len {0}, got {1}")]
+    SrtpMasterKeyLength(usize, usize),
+    #[error("SRTP Salt must be len {0}, got {1}")]
+    SrtpSaltLength(usize, usize),
```

Comment:
I'm very sceptical how user can perform error handling with varying input, I'd suggest we instead make this `SrtpSaltLength` so that library users can easily compare errors. Also for other definitions.

Reply:
I think it should be fine. If users want to compare error, they can use std::matches to ignore usize, https://doc.rust-lang.org/std/macro.matches.html

Reply:
Crates like serde and chrono add data to their error variants, it seems like a common pattern to me.